### PR TITLE
Move ngraph function to new api

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/tests/ngraph_helpers/ngraph_functions/include/ngraph_functions/subgraph_builders.hpp [hunk 1]
```diff
@@ -4,301 +4,456 @@
...
+#include "openvino/op/tensor_iterator.hpp"
+#include "openvino/op/tile.hpp"
 
 namespace ngraph {
 namespace builder {
 namespace subgraph {
-inline std::shared_ptr<ngraph::Function> makeConvPoolRelu(std::vector<size_t> inputShape = {1, 1, 32, 32},
-                                                          ngraph::element::Type_t ngPrc = ngraph::element::Type_t::f32) {
+inline std::shared_ptr<ov::Model> makeConvPoolRelu(std::vector<size_t> inputShape = {1, 1, 32, 32},
+                                                   ov::element::Type_t ngPrc = ov::element::Type_t::f32) {
```
Comment (ilya-lavrenov) [inline src/tests/ngraph_helpers/ngraph_functions/include/ngraph_functions/subgraph_builders.hpp:34]: can we move implementations to cpp files?
in this case we don't need to keep heavy includes files in `ngraph_functions` headers.

File: src/tests/functional/shared_test_classes/include/shared_test_classes/base/layer_test_utils.hpp [hunk 2]
```diff
@@ -11,6 +11,13 @@
 #include <tuple>
 #include <gtest/gtest.h>
 #include <ngraph/node.hpp>
+#include <ngraph/opsets/opset3.hpp>
+#include <ngraph/opsets/opset4.hpp>
+#include <ngraph/opsets/opset5.hpp>
+#include <ngraph/opsets/opset6.hpp>
+#include <ngraph/opsets/opset7.hpp>
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/opsets/opset9.hpp>
```
Comment (ilya-lavrenov) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/base/layer_test_utils.hpp:20]: do we plan to eliminate all of this? 
having global headers in base header files for tests is overkill

Comment (ilyachur) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/base/layer_test_utils.hpp:20]: It should be done in the process of migration shared layer tests

File: src/tests/ngraph_helpers/ngraph_functions/include/ngraph_functions/subgraph_builders.hpp (hunk 1 above)
Comment (olpipi) [reply src/tests/ngraph_helpers/ngraph_functions/include/ngraph_functions/subgraph_builders.hpp:34]: Hi. I'd like to do it a little later in scope of [CVS-117139](https://jira.devtools.intel.com/browse/CVS-117139)

**Review by ilya-lavrenov — APPROVED** [review]

**Review by p-durandin — APPROVED** [review]

File: src/tests/ngraph_helpers/ngraph_functions/src/non_max_suppression.cpp [hunk 3]
```diff
@@ -39,43 +41,43 @@ std::shared_ptr<ngraph::Node> makeNms(const ngraph::Output<Node>& boxes,
...
-                                                  maxBoxesPrec,
-                                                  thrPrec,
-                                                  maxOutBoxesPerClass,
-                                                  iouThr,
-                                                  scoreThr,
-                                                  softNmsSigma,
-                                                  isCenter,
-                                                  sortResDescend,
-                                                  outType);
+        return makeNms<ov::op::v5::NonMaxSuppression>(boxes,
```
Comment (ilya-lavrenov) [inline src/tests/ngraph_helpers/ngraph_functions/src/non_max_suppression.cpp:70]:
```suggestion
        return makeNms<ov::op::v9::NonMaxSuppression>(boxes,
```

**Pull request merged**